File: challenge-171-easy/Main.java
import java.util.Scanner;


public class Main {

	public static void main(String[] args){
		Scanner scanner = new Scanner(System.in);
		String input = scanner.nextLine();
		
		String[] hexs = input.split(" ");
		for (String hex : hexs) {
			String binary = Integer.toBinaryString(Integer.parseInt(hex, 16));
			String result = fillWithZeros(binary);
			System.out.println(result.replaceAll("1", "X").replaceAll("0", " "));
		}
	}
	
	private static String fillWithZeros(String binary){
		if(binary.length() >= 8)
			return binary;
		else
			return fillWithZeros("0" + binary);
	}
}

File: challenge-171-easy/Main.scala
object Main {

	def main(args :Array[String]){
		printHex("18 3C 7E 7E 18 18 18 18")
	}

	def printHex(input: String) = {
		hexlineToBinary(input) .map { line => line.map { c => c match {
					case '1' => 'X'
					case '0' => ' '
				}
			}
		} .foreach(println)
	}

	def hexlineToBinary(input: String) = {
		(input split " ") .map { s =>
			Integer.toBinaryString(Integer.parseInt(s, 16))
		} .map { line => fillWithZeros(line) }
	}

	def fillWithZeros(input: String) : String = {
		if (input.length() >= 8) input else fillWithZeros("0" ++ input)
	}

}
